lib/rubygems.rb

```diff
@@ -941,7 +941,7 @@ def self.ui
   def self.use_paths(home, *paths)
     paths = nil if paths == [nil]
     paths = paths.first if Array === Array(paths).first
-    self.paths = { "GEM_HOME" => home, "GEM_PATH" => paths }
+    self.paths = { "GEM_HOME" => home, "GEM_PATH" => paths.join(Gem.path_separator) }
```

Comment:
Looks like the origin of this regexp stuff is from jruby/jruby@90985cbc, but the commit message doesn't really help.  It would be nice if methods return a consistent type. :disappointed: